src/network/network.ts

```diff
@@ -140,6 +141,33 @@ export class Network {
     return node;
   }
 
+  /**
+   * Remove the node from the network
+   * @param nodeId
+   */
+  async removeNode(nodeId: string): Promise<void> {
+    const node = this.nodes.get(nodeId);
+    if (!node) {
+      throw new GolemError(`Unable to remove node ${nodeId}. There is no such node in the network`);
```

Comment:
I would adjust the type after you merge the PR with new error structures.

Reply:
Yes, after the merge I will resolve all conflicts and improve it.